lib/shared/pe_status_check.rb

```diff
@@ -30,6 +30,10 @@ class << self
   def get_resource(resource, name)
     name += '.service' if (resource == 'service') && !name.include?('.')
     Puppet::Indirector::Indirection.instance(:resource).find("#{resource}/#{name}")
+  rescue StandardError => e
```

Comment:
`NotImplementedError` is not a subclass of `StandardError`, so we may want add `NotImplementedError` or  it's superclass `ScriptError` in the rescue.